### ISSUE TITLE
Use Thematic Breaks (or Horizontal Rules) as Explicit Section Breaks

### ISSUE THREAD

**Issue by Crissov** [arve0/markdown-it-header-sections#8]
There is currently no way to get this HTML from any Markdown with this plugin:

~~~~ html
<section>
  <h1>Upper-Level Heading</h1>
  <p>upper-level section</p>
  <section>
    <h2>Lower-Level Heading</h2>
    <p>lower-level section</p>
  </section>
  <p>upper-level section again!</p>    <!-- this is impossible right now -->
</section>
~~~~

It would be possible, though, if you treated _thematic breaks_ (`---`, `***` or `___`) as *explicit section breaks*. 

~~~~ markdown
Upper-Level Heading
===================

upper-level section

Lower-Level Heading
-------------------

lower-level section

---

upper-level section again!
~~~~

The three different available types could even be used for slightly different meanings:

- `***` end the current section
- `---` end all sections but the top level, except when the current context is the top level and in that case also start a new section
- `___` plain old `<hr />`

Related discussion from CommonMark Discourse
----

- [Horizontal Rule or Thematic Break](https://talk.commonmark.org/t/horizontal-rule-or-thematic-break/912/9)
- [Explicit Section not possible](https://talk.commonmark.org/t/explicit-section-not-possible/796)
- [Horizontal Rules as Hierarchical Section Breaks etc.](https://talk.commonmark.org/t/horizontal-rules-as-hierarchical-section-breaks-etc/1241)

**Comment by arve0** [comment]
Hi!
Thanks for your contribution, thoroughly description and pointers of earlier discussions. First of, I think this sounds like a good idea. Implementing it should be relatively easy too.

A couple of questions:

1. Should this be the default behavior, or is it better to enable it through an option?
2. Regarding differentiating between the three types of thematic breaks, what is the convention/has other projects landed on the same meaning?

**Comment by Crissov** [comment]
1. 🤷 Your plugin, you decide. For slide presentations, it seems like an established convention, though.
2. I don’t think any other implementation has picked up on any differentiation. For `* * *`, I was thinking of its use in books, while `---` resembles second-level headings in Markdown.

**Comment by Crissov** [comment]
PS: Other conventions would be possible and reasonable, too, of course, e.g.:

- `___` end current section `</section>`
- `***` split section `</section><section>`
- `---` start anonymous section `<section>`

**Comment by arve0** [comment]
> PS: Other conventions would be possible and reasonable, too, of course, e.g.:

Yes. I guess one might need sections without headers, especially for slides. I like the `***` idea, it gives a neat slide format.
